virtual_rainforest/models/abiotic_simple/simple_regression.py

```diff
@@ -0,0 +1,588 @@
+    "vapour_pressure_deficit_min": 0,
+    "vapour_pressure_deficit_max": 10,
+    "soil_moisture_min": 0,
+    "soil_moisture_max": 100,
+    "soil_temperature_min": -10,
+    "soil_temperature_max": 50,
+}
+"""Upper and lower bounds for abiotic variables."""
+
+
+def setup_simple_regression(
+    layer_roles: List[str],
+    data: Data,
+    initial_soil_moisture: float = 50,
+) -> Dict[str, DataArray]:
+    r"""Set up abiotic environment variables.
+
+    This function initialises all abiotic variables as DataArrays filled with NaN,
+    except for soil moisture which has a homogenous value across all soil layers.
+
+    Args:
+        layer_roles: list of layer roles (from top to bottom: above, canopy, subcanopy,
+            surface, soil)
+        data: Data object
+        initial_soil_moisture: initial soil moisture
+
+    Returns:
+        Dict of DataArrays for air temperature [C], relative humidity [-], vapour
+        pressure deficit [kPa], soil temperature [C], atmospheric pressure [kPa],
+        atmospheric :math:`\ce{CO2}` [ppm], soil moisture [-], and surface runoff [mm]
+    """
+
+    output = {}
+
+    # Initialise DataArrays filled with NaN
+    output["air_temperature"] = DataArray(
+        np.full((len(layer_roles), len(data.grid.cell_id)), np.nan),
+        dims=["layers", "cell_id"],
+        coords={
+            "layers": np.arange(0, len(layer_roles)),
+            "layer_roles": ("layers", layer_roles),
+            "cell_id": data.grid.cell_id,
+        },
+        name="air_temperature",
+    )
+
+    # copy and assign new variable names
+    for name in (
+        "relative_humidity",
+        "vapour_pressure_deficit",
+        "soil_temperature",
+        "atmospheric_pressure",
+        "atmospheric_co2",
+        "surface_runoff",
+    ):
+        output[name] = output["air_temperature"].copy().rename(name)
+
+    # The initial soil moisture constant in all soil layers, all other layers set to NaN
+    soil_moisture = output["air_temperature"].copy().rename("soil_moisture")
+    output["soil_moisture"] = soil_moisture.where(
+        soil_moisture.coords["layer_roles"] != "soil",
+        initial_soil_moisture,
+    )
+
+    # TODO until the plant model is ready, these variables are initialised here
+    output["layer_heights"] = xr.concat(
+        [
+            DataArray([[32, 32, 32], [30, 30, 30], [20, 20, 20], [10, 10, 10]]),
+            DataArray(np.full((7, 3), np.nan)),
+            DataArray([[1.5, 1.5, 1.5], [0.1, 0.1, 0.1]]),
+            DataArray([[-0.1, -0.1, -0.1], [-1, -1, -1]]),
+        ],
+        dim="dim_0",
+    )
+    output["layer_heights"] = (
+        output["layer_heights"]
+        .rename({"dim_0": "layers", "dim_1": "cell_id"})
+        .assign_coords(
+            {
+                "layers": np.arange(0, 15),
+                "layer_roles": ("layers", layer_roles),
+                "cell_id": data.grid.cell_id,
+            }
+        )
+    )
+
```

Comment:
This could possibly be shorter but it causes errors to just do all in one go, same with `leaf_area_index` below

Reply:
What exactly is the `.T` for? Doe this assign temperature as an attribute?

Reply:
No - sorry that is obscure. It is a `numpy` property that returns the matrix transpose. It just rotate the 3x15 matrix created by broadcasting the vertical layers into a 15x3 matrix as you had in the original inputs. There are a million ways we could create this matrix in `numpy` but this seemed most compact for now.

Reply:
Ah ok, that makes a lot of sense. Thank you :-)